bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -251,7 +255,7 @@ public Rule add(Rule rule) {
      * @param rule candidate for unique ID
      * @return a rule with UID
      */
-    protected Rule initRuleId(String rUID, Rule rule) {
+    protected @Nonnull Rule initRuleId(String rUID, Rule rule) {
```

Comment:
Yes you are right, but are you sure that this annotation is not used at runtime by the VM to ensure that the passed parameter is not null? For example if we have this method:
public void foo(@Nonnull String a) 
the VM has to check on each call if the param a != null otherwise  we can't remove the null check from the method. If so I'm not sure if there is a need from it at all.